docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md

```diff
@@ -0,0 +1,815 @@
+  apply:
+    component: outerloop-role
+- id: k8s-rolebinding-for-helm
+  apply:
+    component: outerloop-rolebinding
+- id: build-image
+  apply:
+    component: outerloop-build
+- id: k8s-deployment
+  apply:
+    component: outerloop-deployment
+- id: k8s-service
+  apply:
+    component: outerloop-service
+- id: k8s-url
+  apply:
+      component: outerloop-url
+```
+
+### Add the `components`
+Every command above references a `component`, and so we now add components under the `components` section.
+
+We will first add the component referenced by `deploy-db` command.
+
+```yaml
+- name: deploy-db
+  container:
+    image: quay.io/tkral/devbox-demo-devbox
+  attributes:
+    pod-overrides:
+      spec:
+        serviceAccountName: my-go-app
+```
+
+The image used by this container component contains the Helm binary that we can use to deploy the helm chart.
+
+The component is using a `pod-overrides` attribute that will override the service account used by the pod to deploy the helm chart to use the service account (`my-go-app`) we define in this Devfile.
+If we do not do this, the pod will use the `default` service account that does not have the required permissions.
+
+We will now add the remaining components.
+
+```yaml
+- name: outerloop-serviceaccount
+  kubernetes:
+    inlined: |
+      apiVersion: v1
+      kind: ServiceAccount
+      metadata:
+        name: {{RESOURCE_NAME}}
+- name: outerloop-role
+  kubernetes:
+    inlined: |
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: Role
+      metadata:
+        name: {{RESOURCE_NAME}}
+      rules:
+      - apiGroups:
+        - '*'
+        resources:
+        - '*'
+        verbs:
+        - '*'
+- name: outerloop-rolebinding
+  kubernetes:
+    inlined: |
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: RoleBinding
+      metadata:
+        name: {{RESOURCE_NAME}}
+      roleRef:
+        apiGroup: rbac.authorization.k8s.io
+        kind: Role
+        name: {{RESOURCE_NAME}}
+      subjects:
+      - kind: ServiceAccount
+        name: {{RESOURCE_NAME}}
+# This will build the container image before deployment
+- name: outerloop-build
+  image:
+    dockerfile:
+      buildContext: ${PROJECT_SOURCE}
+      rootRequired: false
+      uri: ./Dockerfile
+    imageName: "{{CONTAINER_IMAGE}}"
+# This will create a Deployment in order to run your container image across the cluster.
+# Note that we expose the env vars necessary to connect application with the mongodb service.
+- name: outerloop-deployment
+  kubernetes:
+    inlined: |
+      kind: Deployment
+      apiVersion: apps/v1
+      metadata:
+        name: {{RESOURCE_NAME}}
+      spec:
+        replicas: 1
+        selector:
+          matchLabels:
+            app: {{RESOURCE_NAME}}
+        template:
+          metadata:
+            labels:
+              app: {{RESOURCE_NAME}}
+          spec:
+            containers:
+              - name: {{RESOURCE_NAME}}
+ image: {{CONTAINER_IMAGE}}
```

Comment:
Same point: indentation issues..

Reply:
I am not sure when this got messed up. I remember copy pasting the commands as they are and testing them, and it worked fine for me then.